apps/src/openni_feature_persistence.cpp

```diff
@@ -266,6 +266,10 @@ main(int argc, char** argv)
                "MultiscaleFeaturePersistence class using the FPFH features\n"
             << "Use \"-h\" to get more info about the available options.\n";
 
+  std::string arg;
+  if (argc > 1)
```

Comment:
Might be best to also test that the first arg isn't one of the other options (`-octree_leaf_size`, `-normal_search_radius`, etc) so that giving the device id on the command line is still optional. Checking whether the arg begins with a `-` should be enough.
Same for openni_mobile_server.cpp and openni_voxel_grid.cpp